### ISSUE TITLE
Microdown Stef Roadmap

### ISSUE THREAD

**Issue by Ducasse** [pillar-markup/Microdown#324]

**Comment by Ducasse** [comment]
## Pharo10
- Release MD
- Release BC
## For books:
- make sure that we can write “simple” books in microdown and produce latex
		- case one with conversion to Pillar objects [this is working]
		- case two without conversion
		
- Path to go there
  - should manage inputfile (transofmration)
  - resurrect basic converter from PR to MIC files
  - Try on Pharo with Style
  - check LaTeX export 

## For Slides
- support meta data
- introduce columns
- introduce slides
- Latex output (nearly done in fact)
- HTML output

**Comment by kasperosterbye** [comment]
Wrt columns, take a look at the `MicTextPresenter class>>exampleCheetSheet`. It is there already.

**Comment by Ducasse** [comment]
Ok I will have a look 
I wanted to know also how we manage finally the new line in text.
Because there are strange code in some rendering (like splitting text on newline to iterate and to put it back).
S

**Comment by kasperosterbye** [comment]
Yes, that is the meta issues "Several issues is related to handling of children of nodes”. Listed in #323.

It is not a hard issue to address implementation wise, but it has some overall design considerations which it would be useful to discuss.

Best,

Kasper

**Comment by Ducasse** [comment]
Ok what I started to do is to follow the design of the dom or XML
- Domain nodes are always in children and can be optional accessible via specific setters and getter. It means that we can access all the nodes using children.
- Entities that are not in the tree are annotation (for example url of a link).

Is what I'm saying clear?

**Comment by Ducasse** [comment]
The question is do we want the body of a code block (basically a large text or a list of line expressed as a Text elements) to be in the tree or we cut the flow at the block level.

**Comment by kasperosterbye** [comment]
I can live with the XML notion of children/elements, and the the attributes as annotations (typically using MicArgumentSplitter).
But XML is not in agreement on this - this short discussion shows it pretty well: https://www.w3schools.com/xml/xml_dtd_el_vs_attr.asp

**Comment by kasperosterbye** [comment]
But yes - I like

> * Domain nodes are always in children and can be optional accessible via specific setters and getter. It means that we can access all the nodes using children.
>*  Entities that are not in the tree are annotation (for example url of a link).

If that means the variable `children` should be used, or the method `children` should return the thing you ask for - I am perhaps more to the latter version.

**Comment by Ducasse** [comment]
At least we should be able to encapsulate the choices via messages.

**Comment by kasperosterbye** [comment]
Agreed

**Comment by kasperosterbye** [comment]
It seems like 90% of this is done, and 10% is obsolete. So I suggest this to be closed now.
